presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleTester.java

```diff
@@ -45,10 +48,11 @@ public RuleTester()
 
         this.metadata = queryRunner.getMetadata();
         this.costCalculator = queryRunner.getCostCalculator();
+        this.lookup = new TestingLookup(new CoefficientBasedCostCalculator(queryRunner.getMetadata()), ImmutableMap.of());
```

Comment:
I do not like the idea that `costCalculator` in `queryRunner` (`LegacyCostCalculator`) and cost computation logic in memo passed to `RuleAssert` are detached.
Maybe both `LegacyCostCalculator` and `Lookup` implementation should come from `QueryRunner`. So they are constructed in the same place and it is obvious that one matches the other (i.e. Both are based on the same implementation of new `CostCalculator`).